### PR TITLE
added sanity test for updating vector in HNSW algo

### PR DESCRIPTION
**Pull request by GuyAv46**
As a follow up for @meiravgri PR that found a bug in brute force algorithm (#178), I add a test for checking that HNSW move blobs correctly on updates

### CODE REVIEW

File: tests/unit/test_hnswlib.cpp [hunk 1]
```diff
@@ -39,6 +39,49 @@ TEST_F(HNSWLibTest, hnswlib_vector_add_test) {
...
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 1);
+    v = reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getDataByInternalId(0);
+    ASSERT_FALSE(memcmp(v, a, sizeof(a)));
+
+    VecSimIndex_AddVector(index, (const void *)b, 1);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 2);
+    v = reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getDataByInternalId(1);
+    ASSERT_FALSE(memcmp(v, b, sizeof(b)));
+
+    // After inserting c with label 0, we first delete id 0 and move the last id (1) to be 0.
```
Comment (meiravgri) [inline tests/unit/test_hnswlib.cpp:73]: consider checking also updating the last vector

**Comment by meiravgri** [comment]
cool :)

File: tests/unit/test_hnswlib.cpp [hunk 2]
```diff
@@ -39,6 +39,65 @@ TEST_F(HNSWLibTest, hnswlib_vector_add_test) {
...
     VecSimIndex_Free(index);
 }
+
+TEST_F(HNSWLibTest, hnswlib_blob_sanity_test) {
+    size_t dim = 4;
+    size_t bs = 1;
+#define ASSERT_HNSW_BLOB_EQ(id, blob)                                                              \
+    do {                                                                                           \
+        void *v = reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getDataByInternalId(id);   \
+        ASSERT_FALSE(memcmp(v, blob, sizeof(blob)));                                               \
```
Comment (DvirDukhan) [inline tests/unit/test_hnswlib.cpp:49]: Is sizeof(blob) return the actual length or the size of the pointer?

Comment (GuyAv46) [reply tests/unit/test_hnswlib.cpp:49]: Size of the entire blob in bytes

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**